app/controllers/api/metric_visibilities_controller.rb

```diff
@@ -10,33 +10,56 @@ class Api::MetricVisibilitiesController < FrontendController
   def toggle_visible
     @metric.toggle_visible_for_plan(@plan)
 
+    errors = @metric.errors.presence
```

Comment:
It's a bit hard to understand these changes... 

I guess you're replacing what was previously implemented in `js.erb` with plain controller logic?

Reply:
Yes it's basically removing jQuery from those js.erb and moving the logic to server-side instead of doing stuff in the client.